Add dead letter queue to API stack, to limit retries

suzuki-safie opened the pull request:
*Issue #, if available:*

None

*Description of changes:*

When published API handler failed, message in SQS queue will remain in the queue and retry occurs every 30 minutes for 4 days (192 times). This can cause unwanted mass usage of foundation model.

This PR adds dead-letter-queue to the queue and limit max retry count to 1.

* [`cdk/lib/api-publishment-stack.ts`](diffhunk://#diff-f8b13033775c259ddaae6a6fa8b26bd3262b34b9523a4fb3e5db3e9a60b2eb59R36-R44): Added a dead-letter queue `chatQueueDLQ` with a retention period of 14 days and configured the main `chatQueue` to use this DLQ with a maximum receive count of 2 (one retry).

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

suzuki-safie committed:
Add dead letter queue to API stack, to limit retries

statefb commented on cdk/lib/api-publishment-stack.ts:
```diff
@@ -33,8 +33,15 @@ export class ApiPublishmentStack extends Stack {
 
     const deploymentStage = props.deploymentStage ?? "dev";
 
+    const chatQueueDLQ = new sqs.Queue(this, "ChatQueue-DLQ", {
```
Could you omit "-" because naming rule should be follow PascalCase. [source](https://qiita.com/tmokmss/items/721a99e9a62499d6d54a)

suzuki-safie replied:
changed `ChatQueue-DLQ` to `ChatQueueDlq`

statefb reviewed (commented):
@suzuki-safie Thank you for the PR! Please check a comment (except for this, LGTM)

suzuki-safie committed:
fix DLQ name

Change SQS queue DLQ name according to https://qiita.com/tmokmss/items/721a99e9a62499d6d54a. also changed "DLQ" => "Dlq" because not using capitalize-acronym rule.

statefb reviewed (approved):
LGTM!
